servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/H2ToStH1UtilsTest.java

```diff
@@ -26,32 +26,49 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static io.netty.util.internal.PlatformDependent.hashCodeAscii;
+import static io.servicetalk.buffer.api.CharSequences.caseInsensitiveHashCode;
 import static io.servicetalk.http.api.HttpHeaderNames.ACCEPT_PATCH;
 import static io.servicetalk.http.api.HttpHeaderNames.COOKIE;
-import static io.servicetalk.http.api.HttpHeaderValues.TEXT_PLAIN;
+import static io.servicetalk.http.api.HttpHeaderNames.EXPIRES;
 import static io.servicetalk.http.netty.H2ToStH1Utils.h1HeadersSplitCookieCrumbs;
-import static java.util.Objects.requireNonNull;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasSize;
 
 class H2ToStH1UtilsTest {
 
+    private static final int ARRAY_SIZE_HINT = 16;
+    private static final HttpHeadersFactory H1_FACTORY = new DefaultHttpHeadersFactory(true, true,
+            ARRAY_SIZE_HINT, 0);
+    private static final HttpHeadersFactory H2_FACTORY = new H2HeadersFactory(true, true, ARRAY_SIZE_HINT, 0);
+
+    private static int bucketIndex(int hashCode) {
+        return hashCode & (ARRAY_SIZE_HINT - 1);
+    }
+
     @Test
     void testH1HeadersSplitCookieCrumbsForH1Headers() {
-        testH1HeadersSplitCookieCrumbs(DefaultHttpHeadersFactory.INSTANCE);
+        CharSequence secondHeaderName = EXPIRES;
+        assertThat(bucketIndex(caseInsensitiveHashCode(COOKIE)),
+                equalTo(bucketIndex(caseInsensitiveHashCode(secondHeaderName))));
+        testH1HeadersSplitCookieCrumbs(H1_FACTORY, secondHeaderName);
     }
 
     @Test
     void testH1HeadersSplitCookieCrumbsForH2Headers() {
-        testH1HeadersSplitCookieCrumbs(H2HeadersFactory.INSTANCE);
+        CharSequence secondHeaderName = ACCEPT_PATCH;
+        assertThat(bucketIndex(hashCodeAscii(COOKIE)), equalTo(bucketIndex(hashCodeAscii(secondHeaderName))));
```

Comment:
This assertion could be moved in to the `testH1HeadersSplitCookieCrumbs` as the test is invalid if `secondHeaderName` does not match `COOKIE`.

Reply:
`secondHeaderName` and the hashcode function are different between h1 and h2, therefore they are validated before the shared logic can be applied